src/httpfetch.h

```diff
@@ -135,6 +135,9 @@ u64 httpfetch_caller_alloc_secure();
 // to stop any ongoing fetches for the given caller.
 void httpfetch_caller_free(u64 caller);
 
-// Performs a synchronous HTTP request. This blocks and therefore should
-// only be used from background threads.
-void httpfetch_sync(const HTTPFetchRequest &fetch_request, HTTPFetchResult &fetch_result);
+// Performs a synchronous HTTP request that is interruptible if the current
+// thread is a Thread object. interval is the completion check interval in ms.
+// This blocks and therefore should only be used from background threads.
+// Returned is whether the request completed without interruption.
+bool httpfetch_sync_interruptible(const HTTPFetchRequest &fetch_request,
```

Comment:
as you removed http_fetch_sync, i think we can call this httpfetch_sync which is the standard and have a default value

Reply:
Well, httpfetch_sync still exists, it's just private to httpfetch.cpp now.
I'd have to rename httpfetch_sync to something else. Do you have a suggestion?
In the end, it might be cleaner to keep the names as is.